app/multieditmanager.cpp

```diff
@@ -0,0 +1,178 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "multieditmanager.h"
+#include "mixedattributevalue.h"
+#include "staticfeaturesmodel.h"
+#include "featurelayerpair.h"
+#include "qgsmemoryproviderutils.h"
+#include "qgis.h"
+
+
+MultiEditManager::MultiEditManager( QObject *parent )
+  : QObject( parent )
+  , mModel( std::make_unique<StaticFeaturesModel>() )
+{
+  connect( this, &MultiEditManager::layerChanged, this, &MultiEditManager::createTemporaryLayer );
+  connect( mModel.get(), &StaticFeaturesModel::countChanged, this, &MultiEditManager::geometryChanged );
+}
+
+void MultiEditManager::initialize( const FeatureLayerPair &pair )
+{
+  mModel->populate( { pair } );
+
+  if ( mLayer != pair.layer() )
+  {
+    mLayer = pair.layer();
+    emit layerChanged();
+  }
+}
+
+void MultiEditManager::toggleSelection( const FeatureLayerPair &pair )
+{
+  if ( !pair.isValid() ||
+       ( mLayer && mLayer != pair.layer() ) )
+    return;
+
+  if ( mModel->count() == 0 )
+  {
+    mModel->populate( { pair } );
+    return;
+  }
```

Comment:
Do we need an extra handling if `mModel` is empty? Won't the code below handle this situation?

Reply:
I believe this can go now, it was from before I added an explicit `initialize()` method.